packages/core/src/utils.ts

```diff
@@ -554,14 +558,95 @@ export const isTransparentProxy = async (
   return true
 }
 
+/**
+ * Checks if the passed in proxy contract points to an implementation address which implements the minimum requirements to be
+ * a ChugSplash compatible UUPS proxy.
+ *
+ * @param provider JSON RPC provider corresponding to the current project owner.
+ * @param proxyAddress Address of the proxy contract. Since this is a UUPS proxy, we check the interface of the implementation function.
+ * @returns
+ */
+export const isUUPSProxy = async (
+  provider: providers.Provider,
+  proxyAddress: string
+): Promise<boolean> => {
+  const implementationAddress = await getEIP1967ProxyImplementationAddress(
+    provider,
+    proxyAddress
+  )
+
+  // We don't consider default proxies to be ERC proxies, even though they share the same
+  // interface.
+  if ((await isDefaultProxy(provider, implementationAddress)) === true) {
+    return false
+  }
+
+  // Check if the contract bytecode contains the expected interface
+  const bytecode = await provider.getCode(implementationAddress)
+  if (!(await bytecodeContainsUUPSInterface(bytecode))) {
+    return false
+  }
+
+  // Fetch proxy owner address from storage slot defined by EIP-1967
+  const ownerAddress = await getEIP1967ProxyAdminAddress(
+    provider,
+    implementationAddress
+  )
+
+  // If proxy owner is not a valid address, then proxy type is incompatible
+  if (!ethers.utils.isAddress(ownerAddress)) {
+    return false
+  }
+
+  return true
+}
+
+const bytecodeContainsUUPSInterface = async (
```

Comment:
This function doesn't need to be async, and all calls to this function can remove the `await` keyword